url86119f019a_solzy_hive_parse_tgz-pJ8-org_apache_hadoop_hive_ql_parse_MainJ8/src/org/apache/hadoop/hive/ql/parse/HiveParser_FromClauseParser.java

```diff
@@ -0,0 +1,6078 @@
+		  throw e;
+		}
+
+		finally {
+			// do for sure before leaving
+		}
+		return retval;
+	}
+	// $ANTLR end "lateralView"
+
+
+	public static class tableAlias_return extends ParserRuleReturnScope {
+		CommonTree tree;
+		@Override
+		public CommonTree getTree() { return tree; }
+	};
+
+
+	// $ANTLR start "tableAlias"
+	// FromClauseParser.g:135:1: tableAlias : identifier -> ^( TOK_TABALIAS identifier ) ;
+	public final HiveParser_FromClauseParser.tableAlias_return tableAlias() throws RecognitionException {
+		HiveParser_FromClauseParser.tableAlias_return retval = new HiveParser_FromClauseParser.tableAlias_return();
+		retval.start = input.LT(1);
+
+		CommonTree root_0 = null;
+
+		ParserRuleReturnScope identifier64 =null;
+
+		RewriteRuleSubtreeStream stream_identifier=new RewriteRuleSubtreeStream(adaptor,"rule identifier");
+
+		gParent.msgs.push("table alias"); 
+		try {
+			// FromClauseParser.g:138:5: ( identifier -> ^( TOK_TABALIAS identifier ) )
+			// FromClauseParser.g:139:5: identifier
+			{
+			pushFollow(FOLLOW_identifier_in_tableAlias791);
+			identifier64=gHiveParser.identifier();
+			state._fsp--;
+			if (state.failed) return retval;
+			if ( state.backtracking==0 ) stream_identifier.add(identifier64.getTree());
+			// AST REWRITE
+			// elements: identifier
+			// token labels: 
+			// rule labels: retval
+			// token list labels: 
+			// rule list labels: 
+			// wildcard labels: 
+			if ( state.backtracking==0 ) {
+			retval.tree = root_0;
+			RewriteRuleSubtreeStream stream_retval=new RewriteRuleSubtreeStream(adaptor,"rule retval",retval!=null?retval.getTree():null);
+
+			root_0 = (CommonTree)adaptor.nil();
+			// 139:16: -> ^( TOK_TABALIAS identifier )
+			{
+				// FromClauseParser.g:139:19: ^( TOK_TABALIAS identifier )
+				{
+				CommonTree root_1 = (CommonTree)adaptor.nil();
+				root_1 = (CommonTree)adaptor.becomeRoot((CommonTree)adaptor.create(TOK_TABALIAS, "TOK_TABALIAS"), root_1);
+				adaptor.addChild(root_1, stream_identifier.nextTree());
+				adaptor.addChild(root_0, root_1);
+				}
+
+			}
+
+
+			retval.tree = root_0;
+			}
+
+			}
+
+			retval.stop = input.LT(-1);
+
+			if ( state.backtracking==0 ) {
+			retval.tree = (CommonTree)adaptor.rulePostProcessing(root_0);
+			adaptor.setTokenBoundaries(retval.tree, retval.start, retval.stop);
+			}
+			if ( state.backtracking==0 ) {gParent.msgs.pop(); }
+		}
+
+		catch (RecognitionException e) {
+		  throw e;
+		}
+
+		finally {
+			// do for sure before leaving
+		}
+		return retval;
+	}
+	// $ANTLR end "tableAlias"
+
+
+	public static class fromSource_return extends ParserRuleReturnScope {
+		CommonTree tree;
+		@Override
+		public CommonTree getTree() { return tree; }
+	};
+
+
+	// $ANTLR start "fromSource"
+	// FromClauseParser.g:142:1: fromSource : ( ( Identifier LPAREN )=> partitionedTableFunction | tableSource | subQuerySource ) ( lateralView ^)* ;
+	public final HiveParser_FromClauseParser.fromSource_return fromSource() throws RecognitionException {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 134. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.